backend/src/org/commcare/util/FormDataUtil.java

```diff
@@ -22,13 +23,54 @@ public static String getTitleFromSession(UserSandbox userSandbox,
 
         Pair<EntityDatum, String> datumEntityAndValue =
                 getDatumAndCaseId(sessionCopy, evalContext);
-        if (datumEntityAndValue.first == null || datumEntityAndValue.second == null) {
-            return null;
-        }
-
         EntityDatum entityDatum = datumEntityAndValue.first;
         String value = datumEntityAndValue.second;
 
+        if (entityDatum == null) {
+            if (value == null) {
+                return null;
+            } else {
+                return getCaseName(userSandbox, value);
+            }
+        } else {
+            return loadTitleFromEntity(entityDatum, value, evalContext, sessionCopy, userSandbox);
+        }
+    }
+
+    private static Pair<EntityDatum, String> getDatumAndCaseId(CommCareSession session,
+                                                               EvaluationContext evaluationContext) {
+        String datumValue = null;
+        while (session.getFrame().getSteps().size() > 0) {
+            SessionDatum datum = session.getNeededDatum();
+            if (isCaseIdComputedDatum(datum, datumValue, session.getPoppedStep())) {
```

Comment:
No, there are times when it isn't. Step through it while running the `FormDataUtilTests`